Using auto-downloaded Java 8 toolchain on Windows fails intermittently

### Expected Behavior

Auto-downloaded toolchains should be usable.

### Current Behavior

```
Downloading api.adoptopenjdk.net/v3/binary/latest/8/ga/windows/x64/jdk/hotspot/normal/adoptopenjdk to C:\Users\runneradmin\.gradle\jdks\adoptopenjdk-8-x64-windows.zip
Installed adoptopenjdk-8-x64-windows.zip into C:\Users\runneradmin\.gradle\jdks\jdk8u265-b01
Starting process 'command 'C:\Users\runneradmin\.gradle\jdks\jdk8u265-b01\bin\java.exe''. Working directory: C:\Users\RUNNER~1\AppData\Local\Temp\1602322016793-0 Command: C:\Users\runneradmin\.gradle\jdks\jdk8u265-b01\bin\java.exe -Dfile.encoding=windows-1252 -Duser.country=US -Duser.language=en -Duser.variant -cp C:\Users\RUNNER~1\AppData\Local\Temp\1602322016793-0 JavaProbe
Successfully started process 'command 'C:\Users\runneradmin\.gradle\jdks\jdk8u265-b01\bin\java.exe''

> Task :platform-tooling-support-tests:test FAILED
:platform-tooling-support-tests:test (Thread[Daemon worker,5,main]) completed. Took 6.203 secs.

Closing Git publish repo: D:\a\junit5\junit5\documentation\build\gitPublish
341 actionable tasks: 226 executed, 100 from cache, 15 up-to-date
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':platform-tooling-support-tests:test'.
> Illegal char <
> at index 50: C:\Users\runneradmin\.gradle\jdks\jdk8u265-b01\jre

* Try:
Run with --stacktrace option to get the stack trace. Run with --debug option to get more log output. Run with --scan to get full insights.

* Get more help at help.gradle.org

BUILD FAILED in 6m 58s

Publishing build scan...
https://ge.junit.org/s/iqglklqwr7kqq
```

Source: https://github.com/junit-team/junit5/runs/1235021599

### Context

So far we've only seen this issue on Windows with GitHub Actions.

### Steps to Reproduce 

Unfortunately, the issue is intermittent and thus cannot be reproduced reliably.

### Your Environment

Build scan URL: https://ge.junit.org/s/iqglklqwr7kqq

--- a/subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/JavaInstallationProbe.java
+++ b/subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/JavaInstallationProbe.java
@@ -354,7 +354,7 @@ public class JavaInstallationProbe {
         EnumMap<SysProp, String> result = new EnumMap<SysProp, String>(SysProp.class);
         for (SysProp type : SysProp.values()) {
             if (type != SysProp.Z_ERROR) {
-                result.put(type, split[type.ordinal()]);
+                result.put(type, split[type.ordinal()].trim());
             }
         }
         return result;

--- a/subprojects/platform-jvm/src/test/groovy/org/gradle/jvm/toolchain/internal/JavaInstallationProbeTest.groovy
+++ b/subprojects/platform-jvm/src/test/groovy/org/gradle/jvm/toolchain/internal/JavaInstallationProbeTest.groovy
@@ -83,6 +83,7 @@ class JavaInstallationProbeTest extends Specification {
         if (expectedResult == IS_JDK || expectedResult == IS_JRE) {
             assert install.javaVersion == javaVersion
             assert displayName == null || install.displayName == displayName
+            assert probeResult.javaHome != null
         }
 
         where:
@@ -118,12 +119,13 @@ class JavaInstallationProbeTest extends Specification {
         'zuluJdk8'                            | zuluJvm('8')     | JavaVersion.VERSION_1_8 | 'Zulu JDK 8'   | true | false | IS_JDK
         'hpuxJre6'                            | hpuxJvm('6')     | JavaVersion.VERSION_1_6 | 'HP-UX JRE 6'  | true | true  | IS_JRE
         'hpuxJdk7'                            | hpuxJvm('7')     | JavaVersion.VERSION_1_7 | 'HP-UX JDK 7'  | true | false | IS_JDK
+        'whitespaces'                         | whitespaces('11.0.3')  | JavaVersion.VERSION_11  | 'AdoptOpenJDK JRE 11' | true   | true  | IS_JRE
         'binary that has invalid output'      | invalidOutput()  | null                    | null           | true | false | INVALID_JDK
         'binary that returns unknown version' | invalidVersion() | null                    | null           | true | false | INVALID_JDK
     }
 
     @Requires(TestPrecondition.SYMLINKS)
-    def "cached probe are not affecyed by symlink changes"() {
+    def "cached probe are not affected by symlink changes"() {
         given:
         NativeServicesTestFixture.initialize()
         def execFactory = Mock(ExecActionFactory)
@@ -182,6 +184,17 @@ class JavaInstallationProbeTest extends Specification {
          'java.vm.name': "OpenJDK 64-Bit Server VM",
          'java.vm.version': "${version}+7",
          'java.runtime.name': "OpenJDK Runtime Environment"
+        ]
+    }
+
+    private static Map<String, String> whitespaces(String version) {
+        ['java.home': "home-with-whitespaces\r",
+         'java.version': version,
+         'java.vendor': "AdoptOpenJDK\r",
+         'os.arch': "x86_64\r",
+         'java.vm.name': "OpenJDK 64-Bit Server VM\r",
+         'java.vm.version': "${version}+7\r",
+         'java.runtime.name': "OpenJDK Runtime Environment\r"
         ]
     }
 

